azureclient/credentialsawarefactory/spec.go

```diff
@@ -0,0 +1,37 @@
+package credentialsawarefactory
+
+import (
+	"context"
+
+	"github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2019-07-01/compute"
+	"github.com/Azure/azure-sdk-for-go/services/dns/mgmt/2018-05-01/dns"
+	"github.com/Azure/azure-sdk-for-go/services/network/mgmt/2019-11-01/network"
+	"github.com/Azure/azure-sdk-for-go/services/resources/mgmt/2019-05-01/resources"
+	"github.com/Azure/azure-sdk-for-go/services/storage/mgmt/2019-04-01/storage"
+	"github.com/giantswarm/apiextensions/v3/pkg/apis/provider/v1alpha1"
+)
+
+// The credentialsawarefactory.Interface interface defines an interface for retrieving an Azure API Client
+// Based on the Giantswarm Cluster ID. Based on the implementation, it can return Azure API Clients for the
+// Management cluster of the Workload Cluster.
+type Interface interface {
```

Comment:
Like with `basic.Factory`, what if this package was `credentialsaware` and interface `Factory`?

Reply:
same as earlier, how would you name the implementation?